unikraft/lib/template/template.go

```diff
@@ -0,0 +1,262 @@
+	ProvideMain       bool
+	WithDocs          bool
+	WithPatchedir     bool
+	GitInit           bool
+	InitialBranch     string
+	CopyrightHolder   string
+	Year              int
+	Commit            string
+	OriginUrl         string
+
+	KconfigDependencies []string
+	SourceFiles         []string
+}
+
+type TemplateOption func(*Template)
+
+func NewTemplate(ctx context.Context, topts ...TemplateOption) (Template, error) {
+	var templ Template
+
+	for _, topt := range topts {
+		topt(&templ)
+	}
+
+	return templ, nil
+}
+
+// Generate template using `.tmpl` files and `Template` struct fields.
+func (t Template) Generate(ctx context.Context, workdir string) error {
+	t.Year = time.Now().Year()
+	t.Commit = "Initial commit (blank)"
+	if !strings.HasSuffix(workdir, "/") {
+		workdir += "/"
+	}
+
+	// Parsing all the templates.
+	codingStyleTmpl, err := template.New("CondingStyle").Parse(CodingStyleTemplate)
+	if err != nil {
+		return err
+	}
+
+	configUkTmpl, err := template.New("ConfigUk").Parse(ConfigUkTemplate)
+	if err != nil {
+		return err
+	}
+
+	contributingTmpl, err := template.New("Contributing").Parse(ContributingTemplate)
+	if err != nil {
+		return err
+	}
+
+	copyingTmpl, err := template.New("Copying").Parse(CopyingTemplate)
+	if err != nil {
+		return err
+	}
+
+	readmeTmpl, err := template.New("Readme").Parse(ReadmeTemplate)
+	if err != nil {
+		return err
+	}
+
+	makefileUkTmpl, err := template.New("Makefile").Parse(MakefileUkTemplate)
+	if err != nil {
+		return err
+	}
+
+	// Creating projectName directory to store all the template files.
+	libDir := path.Join(workdir, t.ProjectName)
+	if err = os.Mkdir(libDir, os.ModePerm); err != nil {
+		return err
+	}
+
+	// Creating template files to store template text.
+	codingStyleFile, err := os.Create(path.Join(libDir, "CODING_STYLE.md"))
+	if err != nil {
+		return err
+	}
+
+	configUkFile, err := os.Create(path.Join(libDir, "Config.uk"))
+	if err != nil {
+		return err
+	}
+
+	contributingMdFile, err := os.Create(path.Join(libDir, "CONTRIBUTING.md"))
+	if err != nil {
+		return err
+	}
+
+	copyingMdFile, err := os.Create(path.Join(libDir, "COPYING.md"))
+	if err != nil {
+		return err
+	}
+
+	readmeMdFile, err := os.Create(path.Join(libDir, "README.md"))
+	if err != nil {
+		return err
+	}
+
+	makefileUkfile, err := os.Create(path.Join(libDir, "Makefile.uk"))
+	if err != nil {
+		return err
+	}
+
+	// Executing Templates with Template struct values
+	if err = codingStyleTmpl.Execute(codingStyleFile, t); err != nil {
+		return err
+	}
+
+	if err = configUkTmpl.Execute(configUkFile, t); err != nil {
+		return err
+	}
+
+	if err = contributingTmpl.Execute(contributingMdFile, t); err != nil {
+		return err
+	}
+
+	if err = copyingTmpl.Execute(copyingMdFile, t); err != nil {
+		return err
+	}
+
+	if err = readmeTmpl.Execute(readmeMdFile, t); err != nil {
+		return err
+	}
+
+	if err = makefileUkTmpl.Execute(makefileUkfile, t); err != nil {
+		return err
+	}
+
+	if t.ProvideMain {
```

Comment:
```suggestion
	if t.ProvideCMain {
```

I would specify that the Main file is actually C

Reply:
Done 👍🏽